### PR TITLE
handlers don #10

### PR DESCRIPTION
**Pull request by alMutazBeAllah**
Create publicHandler
  Create homeHandler
  Create authHandler
  Create errorHandler
  Create serverErrorHandler()

### CODE REVIEW

File: src/server/handlers.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        const filePath = path.join(__dirname, '..', '..', 'public', request.url);
+        fs.readFile(filePath, (error, file) => {
+            if(error){
+                serverHandler();
+            }
+            response.writeHead(200, {'Content-Type' : contentTypeMapping[extention]});
+            response.end(file);
+        });
+    };
+};
```
Comment (mhmdtshref) [inline src/server/handlers.js]: Add `module.exports={ ... }` here.

File: src/server/handlers.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+// -- Error 404 -------------------------
+const errorHandler = (request, response) => {
+    response.writeHead(404, {'Content-Type' : 'text/html'});
+    response.end('<h1>Page Not Found : Error 404</h1>');
+};
+
+// -- Error 500 -------------------------
+const serverHandler = (request, response) => {
+    response.writeHead(505, {'Content-Type' : 'text/html'});
```
Comment (mattlub) [inline src/server/handlers.js]: 505 is not the correct status code:
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_Server_errors

**Comment by mattlub** [comment]
hint: if you add "fixes #10" to the description of the PR, then when the PR is merged, the issue will be closed automatically

File: src/server/handlers.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// -- required --------------------------
+const path = require('path');
+const fs = require('fs');
+const URL = require('url');
```
[the remark this answers is not in the document]
Comment (alMutazBeAllah) [reply src/server/handlers.js]: thank you . I did it

File: src/server/handlers.js (hunk 2 above)
Comment (alMutazBeAllah) [reply src/server/handlers.js]: thank you. I did it.

File: src/server/handlers.js (hunk 1 above)
Comment (alMutazBeAllah) [reply src/server/handlers.js]: thank you , I did it.

**Pull request merged**